### ISSUE TITLE
Bio Metric - Response Ratio for Menstrual Cycle Survey

### ISSUE THREAD

**Issue by HopkinsDC** [Analyticsphere/metricsReportsRequests#21]
Please run a _one-time_ metric for the response rate for the next menstrual cycle survey for those eligible by site, one for research (BioSpm_Setting_v1r0; 650516960=0) and one for clinical (BioSpm_Setting_v1r0; 650516960=1). 

The denominator is all participants eligible for the menstrual cycle survey (SrvMC_MenstSrvElig_v1r0; 289750687=1). The numerator is participants with a menstrual cycle survey completion status of Submitted (SrvMC_BaseComplete_v1r0; 459098666=2). 

See the enable for table below:
- The table should have column percents for "Eligible", "Not Started", "Started", and "Submitted". 
- The "Response Rate" column should be percent (# Submitted/# Eligible)
- Percents in the table should have 1 decimal place
- The response rate in the "Total" row should be the overall rate for all eligible at all sites collectively (Total # Submitted/Total # Eligible)
![Screenshot 2023-04-06 at 2 11 24 PM](https://user-images.githubusercontent.com/85250133/230461166-38d33b52-f2ab-475a-966c-0d279a858577.png)

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  Since SrvMC_MenstSrvElig_v1r0 (d_289750687) is a new variable it'll probably make more sense for me to keep the original eligibility requirements of  (SrvBLM_ResSrvCompl_v1r0 OR SrvBlU_BaseComplete_v1r0) submitted and SrvBlU_MENST60_v1r0=yes, right? Or did you specifically want to test this new derived variable?

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  The data is also currently showing that only research sites have eligible participants. Does that sound possible?

I filtered instead by site, (as d_650516960 is only in the bioSurvey table and there are only 13 Connect_IDs in that table) and according to the menstrual cycle survey only those in HP, Henry Ford, Marshfield, Sanford, and UC have become eligible

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 

Please the following for "Eligible for menstrual cycle survey" demoniator:

SrvBLM_ResSrvCompl_v1r0 (299215535.265193023) = submitted (231311385) AND SrvBlU_MENST60_v1r0 (112151599) = YES (353358909) 
OR 
SrvBlU_BaseComplete_v1r0 (826163434.253883960) = submitted (231311385) AND SrvBlU_MENST60_v1r0 (112151599) = YES

Let me know if you have the same problem with only seeing research participants.

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  That's the logic that I have in place now and I'm still seeing that problem

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 Ok, weird. Can you just use SrvMC_BaseComplete_v1r0 (459098666) instead and instead of the Eligible column, have a total column at the end?

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC So I was able to do the Research table the way you wanted. Does this still work?
![Research Menstrual Cycle Survey Completeness and Response](https://user-images.githubusercontent.com/101136647/231270141-03826a16-1713-41a5-9aed-1223fb0b565e.PNG)

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 Yes, this looks like it'll work. 

Please make the following changes:
- add the count to each cell along with the percentage
- change to row percents
- remove the Total row
- change the Eligibility column to a Total column and move it to be between Submitted and Response Rate.
- calculate the Response Rate as the row Submitted over the row Total.

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC For that case wouldn't the response rate and the submitted values be the same? Submitted per site over Total eligible?
![Menstrual Research 2](https://user-images.githubusercontent.com/101136647/231486478-da8a0767-3a23-4c20-bb68-ff7300b3162c.PNG)

**Comment by KELSEYDOWLING7** [comment]
Or should the total still be the same values as eligible before, and then response rate be submitted over all participants of that site, eligible or not?

**Comment by HopkinsDC** [comment]
Oh, you're right! Yes, the total should be "Total Eligible", so whatever it was before, but just the count for the site; exclude the percent from that row only.

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC 

Got it! Here's the final outcome then 
![Research Menstrual Cycle Survey Completeness and Response](https://user-images.githubusercontent.com/101136647/231493910-85716c79-0aef-4fd0-b2de-1b4c4d9a6bce.PNG)

**Comment by HopkinsDC** [comment]
Thank you, Kelsey!

@brotzmanmj Michelle, I'm unfamiliar with the eligibility criteria for the menstrual cycle survey, so does the table Kelsey generated reflect what you would expect as far as the Total Eligible is concerned? If not, please advise on what variable(s) should be used for this metric. Thanks!

**Comment by brotzmanmj** [comment]
(also there are a few spelling errors in the title of the table above, if you can fix those on the next run too, thanks)

**Comment by KELSEYDOWLING7** [comment]
@brotzmanmj  Yes, just confirmed the figures are correct. And sorry about that, p
![Research Menstrual Cycle Survey Completeness and Response](https://user-images.githubusercontent.com/101136647/231521322-efeb230d-71bd-48ea-ad95-4487ca2a7de6.PNG)
lease see the updated table below

**Comment by KELSEYDOWLING7** [comment]
@brotzmanmj   right now according to the data the only sites (d_827220437) that have data with the menstrual cycle survey requirements are HP, Henry Ford, Marshfield, Sanford Health, and UC. None of the KP Sites.

**Comment by brotzmanmj** [comment]
@KELSEYDOWLING7 are you saying that we're not actually triggering the menstrual cycle survey following clinical collections yet? Can you confirm with Davin and Tony on this (or have you already)? thanks

**Comment by KELSEYDOWLING7** [comment]
@brotzmanmj  No I don't think that's quite it. I'm saying as of right not KPs have no one eligible for the menstrual cycle survey just yet

**Comment by brotzmanmj** [comment]
Got it, OK thanks!

**Comment by KELSEYDOWLING7** [comment]
@brotzmanmj No problem! Let me know how this looks.
![Menstrual Cycle](https://user-images.githubusercontent.com/101136647/231567188-3671148f-889b-49b2-b40d-88bb5c536159.PNG)

**Comment by brotzmanmj** [comment]
Great, thank you! The only last thing I would change if it's possible to do so is to put the percentages in parentheses instead of separately but a comma, just because it's easier to read.

**Comment by KELSEYDOWLING7** [comment]
Sure thing!
![Menstrual Cycle](https://user-images.githubusercontent.com/101136647/231574763-e4469743-95d1-413d-9bc5-44266559322f.PNG)

**Comment by brotzmanmj** [comment]
Terrific, thank you! We should add this to the survey completion metrics that we monitor weekly. I could see this fitting in either on the Biospecimen Weekly Metrics or on the Weekly Module Metrics where we monitor completion of all the other survey modules.

**Comment by KELSEYDOWLING7** [comment]
No problem! Ok, just let me know which you prefer it on and I can add it. I agree that that it would fit nicely with the Weekly Module Metrics with all other surveys.

**Comment by brotzmanmj** [comment]
Sounds good, let's put it there

**Comment by KELSEYDOWLING7** [comment]
Ok, that's now number 8
